File: app/src/main/java/be/appfoundry/aipdemo/injection/AppComponent.java
package be.appfoundry.aipdemo.injection;

import javax.inject.Singleton;

import be.appfoundry.aipdemo.activity.dagger2.Dagger2Activity;
import be.appfoundry.aipdemo.activity.dbflow.DBFlowActivity;
import be.appfoundry.aipdemo.activity.mvp.MVPActivity;
import be.appfoundry.aipdemo.activity.retrofit.RetrofitActivity;
import be.appfoundry.aipdemo.activity.rxjava.ReactiveActivity;
import be.appfoundry.aipdemo.mvp.MVPPresenterDBFlowImpl;
import be.appfoundry.aipdemo.mvp.MVPPresenterRetrofitImpl;
import dagger.Component;

@Singleton
@Component(modules = {AppModule.class, ServiceModule.class})
public interface AppComponent {

    void inject(Dagger2Activity activity);
    void inject(RetrofitActivity activity);
    void inject(DBFlowActivity activity);
    void inject(MVPActivity activity);
    void inject(ReactiveActivity activity);
    void inject(MVPPresenterRetrofitImpl presenter);
    void inject(MVPPresenterDBFlowImpl presenter);
}

File: app/src/main/java/be/appfoundry/aipdemo/injection/AppComponent.kt
package be.appfoundry.aipdemo.injection

import be.appfoundry.aipdemo.activity.dagger.DaggerActivity
import be.appfoundry.aipdemo.activity.retrofit.RetrofitActivity
import be.appfoundry.aipdemo.activity.rxjava.ReactiveActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(
    modules = arrayOf(
        AppModule::class,
        ServiceModule::class
    )
)
interface AppComponent {
  fun inject(target: DaggerActivity)
  fun inject(target: RetrofitActivity)
  fun inject(target: ReactiveActivity)
}